backend/server.py

```
@@ -44,6 +45,27 @@
+    }), e.code
+
+@app.errorhandler(Exception)
+def handle_unexpected_exception(e):
+    return jsonify({
+        "error": "Internal server error",
+        "type": e.__class__.__name__,
+    }), 500
+
+def require_json_field(data, field_name):
+    if not data:
+        raise BadRequest("Invalid or missing JSON body")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)